File: c/difference-of-squares/difference_of_squares.c
#include "difference_of_squares.h"


unsigned int sum_of_squares(unsigned int number)
{
  unsigned int s = 0;
  for (unsigned int i = 1; i <= number; ++i)
    {
      s += i * i;
    }
  return s;
}

unsigned int square_of_sum(unsigned int number)
{
  unsigned int s = 0;
  for (unsigned int i = 1; i <= number; ++i)
    {
      s += i;
    }
  return s * s;
}

unsigned int difference_of_squares(unsigned int number)
{
  return square_of_sum (number) - sum_of_squares (number);
}

File: lua/difference-of-squares/difference-of-squares.lua
local function square_of_sum(n)
	local gauss = ((n + 1) * n) / 2
	return gauss ^ 2
end

local function sum_of_squares(n)
	return ((2 * n + 1) * n * (n + 1)) / 6
end

local function difference_of_squares(n)
	return square_of_sum(n) - sum_of_squares(n)
end

return { square_of_sum = square_of_sum, sum_of_squares = sum_of_squares, difference_of_squares = difference_of_squares }
